prow/external-plugins/cherrypicker/server.go

```diff
@@ -269,9 +272,28 @@ func (s *Server) handlePullRequest(l *logrus.Entry, pre github.PullRequestEvent)
 		}
 		requestorToComments[c.User.Login][targetBranch] = &c
 	}
+
+	// now look for our special labels
+	labels, err := s.ghc.GetIssueLabels(org, repo, num)
+	if err != nil {
+		return err
+	}
+
+	if requestorToComments[pr.User.Login] == nil {
+		requestorToComments[pr.User.Login] = make(map[string]*github.IssueComment)
+	}
+
+	labelPrefix := "cherrypick/"
+	for _, label := range labels {
+		if strings.HasPrefix(label.Name, labelPrefix) {
+			requestorToComments[pr.User.Login][label.Name[len(labelPrefix):]] = nil // leave this nil which indicates a label-initiated cherry-pick
+		}
+	}
+
 	if len(requestorToComments) == 0 {
```

Comment:
We will never hit this case now because we prematurely add an entry for `pr.User.Login` even if there are no `cherrypick/*` labels.

Reply:
Good catch. Fixed.